sale_commission_product_criteria/models/commission.py

```diff
@@ -0,0 +1,392 @@
+        readonly=False,
+        copy=True,
+    )
+
+    def _get_commission_items(self, commission, product):
+        # Method replaced
+        categ_ids = {}
+        categ = product.categ_id
+        while categ:
+            categ_ids[categ.id] = True
+            categ = categ.parent_id
+        categ_ids = list(categ_ids)
+        # Select all suitable items. Order by best match
+        # (priority is: all/cat/subcat/product/variant).
+        self.env.cr.execute(
+            """
+            SELECT
+                item.id
+            FROM
+                commission_item AS item
+            LEFT JOIN product_category AS categ ON item.categ_id = categ.id
+            WHERE
+                (item.product_tmpl_id IS NULL OR item.product_tmpl_id = any(%s))
+                AND (item.product_id IS NULL OR item.product_id = any(%s))
+                AND (item.categ_id IS NULL OR item.categ_id = any(%s))
+                AND (item.commission_id = %s)
+            ORDER BY
+                item.applied_on, categ.complete_name desc, item.id desc
+            """,
+            (
+                product.product_tmpl_id.ids,
+                product.ids,
+                categ_ids,
+                commission._origin.id,  # Added this
+            ),
+        )
+        item_ids = [x[0] for x in self.env.cr.fetchall()]
+        return item_ids
+
+    def _get_single_commission_amount(self, commission, subtotal, product, quantity):
+        self.ensure_one()
+        item_ids = self._get_commission_items(commission, product)
+        if not item_ids:
+            return 0.0
+        commission_item = self.env["commission.item"].browse(item_ids[0])
+        if commission.amount_base_type == "net_amount":
+            # If subtotal (sale_price * quantity) is less than
+            # standard_price * quantity, it means that we are selling at
+            # lower price than we bought, so set amount_base to 0
+            subtotal = max([0, subtotal - product.standard_price * quantity])
+        self.applied_commission_item_id = commission_item
+        # if self.agent_id.use_multi_type_commissions:
+        self.applied_commission_id = commission_item.commission_id
+        if commission_item.commission_type == "fixed":
+            return commission_item.fixed_amount
+        elif commission_item.commission_type == "percentage":
+            return subtotal * (commission_item.percent_amount / 100.0)
+
+
+class SaleCommission(models.Model):
+    _inherit = "sale.commission"
+
+    commission_type = fields.Selection(
+        selection_add=[("rules", "Rules")], ondelete={"rules": "cascade"}
+    )
+    item_ids = fields.One2many("commission.item", "commission_id")
+    sequence = fields.Integer(
+        "Sequence", default=1, help="The first in the sequence is the default one."
+    )
+
+    def write(self, vals):
+        res = super(SaleCommission, self).write(vals)
+        self.sort_items()
+        return res
+
+    def sort_items(self):
```

Comment:
Why not putting this through `_order` in the model?

Reply:
it will be inherited in next modules and logic here will be more compex.

Reply:
In this SQL:

https://github.com/OCA/commission/pull/414/files#diff-ae96df1a277be52a5924c783ffd66ee5aa73b373f4b090f66770d639139d7ba8R36

add the criteria for commission_id.active = True.

By the way, you are not filtering right now by item.active either.

Reply:
ok got it. updated.